portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,39 @@
 
 package com.google.sps.servlets;
 
+import com.google.gson.Gson;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  // comments data structure
+  ArrayList<String> list = new ArrayList();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Steph!</h1>");
+    Gson gson = new Gson();
+    String json = gson.toJson(list);
+
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      String comment = request.getParameter("text-input");
+      if (comment == null) {
+        comment = "";
+      }
+      System.out.println(comment);
```

Comment:
Logging on the server is much more common and can be extremely useful for debugging and monitoring a live app. If you want to keep this one, I would suggest adding some extra info to the log like the request verb, url path, user agent, etc.

There are out-of-the-box solutions to this (request -> pretty log), but you could look into those in the future.